### DISCUSSION TITLE
The Shell Script does not run on Windows Server IIS

### DISCUSSION THREAD

**Discussion by r-a-j** [ImageMagick/ImageMagick#1882] [General] [answered]
**_ISSUE:-_** https://stackoverflow.com/questions/61058428/shellscript-is-not-executing-on-production-server-when-invoked-with-c-sharp-code

**_Fred's Image Magick Script:-_** http://www.fmwconcepts.com/imagemagick/texteffect/index.php

**Comment by dlemstra** [comment, marked as answer]
This looks like a question about your server setup instead of ImageMagick. I hope you will get an answer on StackOverflow.

**Reply by r-a-j** [reply]
> This looks like a question about your server setup instead of ImageMagick. I hope you will get an answer on StackOverflow.

Okay Thank You Very Much :)

**Comment by fmw42** [comment]
@r-a-j  

My scripts are bash unix shell scripts. They do not run under Windows or C-sharp. They should run under Window 10 unix or Windows w/Cygwin

**Reply by r-a-j** [reply]
@fmw42 Yes I ran them under Cygwin on Windows server. They ran when manually run on bash. But when I executed the bash (with the code to invoke the bash and script) using c# code It was not running  and also not showing up under windows processes. I used (C:\cygwin64\bin\bash --login -c "%allParameters%") to invoke the shell script where %allParameters% refers to ( CurveTextEffect.bat texteffect -t 'USA BALL' -S 700x -s outline -e concave -d 0.5 -f Impact -p 29.7286291125 -c '#FFFFFF'   -l 0.41289762 -b none 'D:/1.0.0/CurvedImage.png' )

**Reply by fmw42** [reply]
Unfortunately, I do not know C# and how it invokes a bash script.  But one thing to try is to include the full path to texteffect in your command line. Also to your image. But I notice that you have only one image. You need an input and an output image in my script command.

**Reply by r-a-j** [reply]
Ummm.. Yaa right I can try giving input image path. But I think this texteffect script generates the image based on the input parameters provided. 
The scripts are working perfectly fine. 
I just need to find that tiny little bug in my commands formation ( .bat file ) while automatic invocation through C# code.

**Reply by fmw42** [reply]
Sorry, you are correct. The script creates the output. There is no input.  The other approach would be to simply use ImageMagick to warp some text image to that shape. That is what I do in the script. I create some text image, then warp it appropriately for each different distortion listed. That avoids the bash issue. Just write the warping commands using -distort xxx equivalent in C#.

**Reply by r-a-j** [reply]
Okay, This might work. I'll try that.
Thanks for this idea Fred. 👍

**Reply by fmw42** [reply]
Let me know if you need help with the warping.

**Reply by r-a-j** [reply]
Okay sure.
